### ISSUE TITLE
Recompile issue typedefs.f90 on CSC Puhti

### ISSUE THREAD

**Issue by eefvandongen** [joeatodd/HiDEM#2]
**Computing system (please complete the following information):**
 - Machine name: Puhti @ CSC, Espoo 
 - HiDEM commit hash [e.g. 53e2eb5]: 924c98339fcbe835d5d2680d954767917baf24e4

**Issue Type**
Error message in compilation

**Error message or symptoms**
mpif-sizeof.h:395.47:
    Included at mpif.h:63:
    Included at /users/eefvan/sources/puhti/HiDEM/src/typedefs.f90:14:

      COMPLEX(REAL64), DIMENSION(1,1,1,1,1,1,1,*)::x
                                               1
Error: Array specification at (1) has more than 7 dimensions
Fatal Error: Error count reached limit of 25.

This same error (more than 7 dimensions) came up 25 times. Maybe related to:
CMake Warning:
  Manually-specified variables were not used by the project:
    CMAKE_TOOLCHAIN_FILE

**Describe the context**
Due to problems on Puhti, I had to recompile HiDEM. I have removed the build directory and tried to use the same compilation file as before, but then the above errors came up.

**Comment by joeatodd** [comment]
I guess this isn't the log from a fresh (rm -r build/*) build? Otherwise, the fact that its not using the toolchain file is probably an issue.

Which toolchain are you using?

**Comment by eefvandongen** [comment]
I did remove the build directory (I tried first not to remove it, but then the recompiled HiDEM still caused problems on puhti so I hoped that a fresh recompilation may solve those problems). Below are the .sh and .cmake (but with changed .txt extension here for Github upload)
[compile_cray.txt](https://github.com/joeatodd/HiDEM/files/3864206/compile_cray.txt)
[HiDEM-cray-taito.txt](https://github.com/joeatodd/HiDEM/files/3864207/HiDEM-cray-taito.txt)

**Comment by joeatodd** [comment]
So, to be totally clear, even when you remove everything from the build directory, your log still warns: 'CMake Warning:
Manually-specified variables were not used by the project:
CMAKE_TOOLCHAIN_FILE' ?

**Comment by eefvandongen** [comment]
yes

**Comment by eefvandongen** [comment]
Indeed, the problem is related to not using the toolchain, which leads to wrong compilers being used and therefore the error messages came up. When I copied the CMake files/folders from the previous builddir to the new builddir, but made sure the builddir was empty besides those copies,  it indeed compiles without errors. So the issue is partly solved, except that it would be good to fix the compilation script such that the right toolchain is automatically used.

**Comment by joeatodd** [comment]
I cannot understand why the toolchain wouldn't be used. I suspect maybe you accidentally have CMakeCache.txt or something in amongst your source files?
